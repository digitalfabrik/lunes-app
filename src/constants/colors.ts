export const COLORS = {
  lunesBlack: '#000e38',
  lunesWhite: '#f9fafb',
  lunesGreyMedium: '#5b616a',
  lunesGreyDark: '#3a4049',
  white: '#ffffff',
  lunesBlackUltralight: '#e0e4ed',
  lunesFunctionalIncorrectDark: '#ff5252',
  lunesRedLight: '#f57f7a',
  lunesGreenLight: '#7ae6d8',
  lunesGreenMedium: '#20c5b4',
  lunesBlueLight: '#97aef1',
  lunesOrangeLight: '#ffbb4a',
  lunesArtikelDas: '#72f399',
  lunesArtikelDiePlural: '#eee12d',
  lunesArtikelDie: '#faa7a7',
  lunesArtikelDer: '#8cc8f3',
  lunesOverlay: 'rgba(0, 14, 56, 0.8)',
  lunesBlackLight: '#777e91',
  blue: '#0000FF',
  lunesFunctionalCorrectDark: '#0adb75',
  lunesFunctionalAlmostCorrectDark: '#ffbb4a',
  lunesRedDark: '#F1635F',
  lunesRed: '#ffa3a3',
  shadow: 'rgba(0, 0, 0, 0.6)',
  lunesBlackMedium: '#3d4662'
}
