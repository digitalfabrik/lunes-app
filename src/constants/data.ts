import {
  easy,
  hard,
  CorrectIcon,
  IncorrectIcon,
  AlmostCorrectIcon,
  CorrectEntriesIcon,
  IncorrectEntriesIcon,
  AlmostCorrectEntriesIcon
} from '../../assets/images'

export const SCREENS = {
  profession: 'Profession',
  professionSubcategory: 'ProfessionSubcategory',
  exercises: 'Exercises',
  vocabularyOverview: 'VocabularyOverview',
  vocabularyTrainer: 'VocabularyTrainer',
  initialSummaryScreen: 'InitialSummary',
  ResultsOverview: 'ResultsOverview',
  CorrectResults: 'CorrectResults',
  ResultScreen: 'ResultScreen',
  AlmostCorrectResults: 'AlmostCorrectResults',
  IncorrectResults: 'IncorrectResults'
}

export const RESULT_TYPE = ['correct', 'similar', 'incorrect']

export const EXERCISES = [
  {
    id: 1,
    title: 'Vocabulary Overview',
    description: 'All Words',
    Level: easy,
    nextScreen: SCREENS.vocabularyOverview
  },
  {
    id: 2,
    title: 'Vocabulary Trainer',
    description: 'Write words with articles',
    Level: hard,
    nextScreen: SCREENS.vocabularyTrainer
  }
]

export const BUTTONS_THEME = {
  light: 'light',
  dark: 'dark'
}

export const RESULTS = [
  {
    id: 1,
    title: 'Correct entries',
    icon: CorrectIcon,
    nextScreen: SCREENS.CorrectResults
  },
  {
    id: 2,
    title: 'Almost correct entries',
    icon: AlmostCorrectIcon,
    nextScreen: SCREENS.AlmostCorrectResults
  },
  {
    id: 3,
    title: 'Incorrect entries',
    icon: IncorrectIcon,
    nextScreen: SCREENS.IncorrectResults
  }
]

export const ARTICLES = {
  die: 'die',
  der: 'der',
  das: 'das',
  diePlural: 'die (plural)'
}

export const RESULT_PRESETS = {
  similar: {
    Icon: AlmostCorrectEntriesIcon,
    title: 'Almost Correct',
    next: { title: 'INCORRECT', type: 'incorrect' }
  },
  correct: {
    Icon: CorrectEntriesIcon,
    title: 'Correct',
    next: { title: 'ALMOST CORRECT', type: 'similar' }
  },
  incorrect: {
    Icon: IncorrectEntriesIcon,
    title: 'Incorrect',
    next: { title: 'CORRECT', type: 'correct' }
  }
}
