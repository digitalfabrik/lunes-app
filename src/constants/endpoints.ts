export const ENDPOINTS = {
  professions: {
    all: '/disciplines'
  },
  subCategories: {
    all: '/training_set/:id'
  },
  documents: {
    all: '/documents/:id'
  }
}
