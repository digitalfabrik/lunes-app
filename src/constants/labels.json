{
  "scopeSelection": {
    "welcome": "Willkommen bei Lunes!",
    "selectProfession": "Wähle den Beruf, den du lernen möchtest.",
    "skipSelection": "Überspringen",
    "confirmSelection": "Starten"
  },
  "home": {
    "welcome": "Willkommen bei Lunes!",
    "haveFun": "Viel Spaß beim Lernen",
    "progressDescription": "Modulen erledigt",
    "start": "Starten",
    "continue": "Fortsetzen",
    "viewDisciplines": "Module ansehen",
    "addCustomDiscipline": "Bereich hinzufügen",
    "customDisciplineSection": "Individueller Lernbereich",
    "customDisciplineExplanation": "Hier kannst du eigene Lernbereiche hinzufügen",
    "errorLoadCustomDiscipline": "Fehler: Bereich konnte nicht geladen werden. Ungültiger Schlüssel:",
    "deleteProfession": "Bereich löschen",
    "impressum": "Impressum"
  },
  "manageSelection": {
    "heading": "Verwalte deine Lernbereiche",
    "yourProfessions": "Deine Berufsauswahl",
    "yourCustomDisciplines": "Deine individuellen Bereiche",
    "yourSavedProgress": "Deine gespeicherten Lernbereiche",
    "addProfession": "Beruf hinzufügen",
    "descriptionAddCustomDiscipline": "Hier kannst du eigene Lernbereiche mit deinem QR-Code hinzufügen.",
    "deleteModal": {
      "confirmationQuestion": "Willst du den Bereich wirklich löschen?",
      "explanation": "Deine Lernfortschritte werden gespeichert und du kannst den Beruf später wiederherstellen.",
      "confirm": "Ja, löschen!"
    }
  },
  "addCustomDiscipline": {
    "heading": "Individuellen Bereich hinzufügen",
    "description": "Bitte gib den Code über die Tastatur ein oder scanne den QR-Code.",
    "placeholder": "Code eingeben",
    "submitLabel": "hinzufügen",
    "backNavigation": "Zur Startseite",
    "error": {
      "alreadyAdded": "Dieser Code wurde bereits erfolgreich hinzufügt.",
      "wrongCode": "Dieser Code existiert nicht (ungültiger Code).",
      "technical": "Ein technischer Fehler ist aufgetreten. Prüfen Sie Ihre Internetverbindung."
    }
  },
  "exercises": {
    "vocabularyList": {
      "title": "Wortliste",
      "description": "Alle Wörter kennenlernen",
      "alternativeWords": "Andere Wörter",
      "suggestAlternative": "Anderes Wort vorschlagen"
    },
    "wordChoice": {
      "title": "Level 1",
      "description": "Das richtige Wort auswählen",
      "errorWrongDisciplineSize": "Die Übung ist für dieses Modul nicht möglich (zu wenig Wörter)"
    },
    "articleChoice": {
      "title": "Level 2",
      "description": "Den richtigen Artikel auswählen"
    },
    "lockedExerciseModal": {
      "title": "Hoppla, dieses Level ist noch nicht freigeschalten.",
      "confirmButtonLabel": "Zu",
      "confirmButtonLabelDeclinated": "Zur",
      "descriptionPart1": "Du musst",
      "descriptionPart2": "abschließen, um das nächste Level freizuschalten."
    },
    "write": {
      "title": "Level 3",
      "description": "Das Wort mit Artikel schreiben",
      "checkInput": "Eingabe überprüfen",
      "showSolution": "Lösung anzeigen",
      "insertAnswer": "Wort mit Artikel eingeben",
      "feedback": {
        "articleMissing": "Ups, deine Eingabe ist nicht vollständig. Füge einen Artikel ein.",
        "wrongWithSolution": "Die richtige Antwort ist:",
        "wrong": "Das war leider nicht richtig.",
        "almostCorrect1": "Deine Eingabe",
        "almostCorrect2": "ist fast richtig.",
        "correct": "Toll, weiter so! \nDeine Eingabe ist richtig."
      }
    },
    "tryLater": "Später versuchen",
    "cheat": {
      "succeed": "Solve All",
      "fail": "Fail All"
    },
    "next": "Nächstes Wort",
    "showResults": "Ergebnisse anzeigen",
    "cancelModal": {
      "cancelAsk": "Möchtest du diese Übung wirklich beenden?",
      "cancel": "Beenden"
    },
    "feedback": {
      "positive": "Weiter so, das war super!",
      "negative": "Das geht besser. Probier es nochmal."
    }
  },
  "results": {
    "feedbackGood": "Toll, weiter so! \nDu hast die Übung sehr gut gemacht.",
    "feedbackBad": "Nicht aufgeben! \nVersuche es noch einmal!",
    "finishedDiscipline": "Super, du hast das Modul abgeschlossen",
    "unlockExercise": {
      "part1": "Glückwunsch",
      "part2": "ist jetzt freigeschaltet."
    },
    "correct": "richtig",
    "entries": "Eingaben",
    "action": {
      "nextExercise": "Nächste Übung",
      "continue": "Weitermachen",
      "repeat": "Übung wiederholen",
      "disciplineOverview": "Zur Modulübersicht",
      "backToWordlist": "Zurück zur Wortliste"
    },
    "of": "von",
    "share": {
      "description": "Teile deine Ergebnisse mit deinen Freunden!",
      "button": "Ergebnisse teilen",
      "message1": "Meine Ergebnisse für",
      "message2": "in Lunes:",
      "message3": "Wörter richtig. Und du?"
    }
  },
  "feedback": {
    "sendFeedback": "Senden",
    "question": "Hast du Feedback oder einen Wortvorschlag?",
    "mailPlaceholder": "Deine Email für Fragen (optional)",
    "feedbackPlaceholder": "Teile dein Feedback hier"
  },
  "general": {
    "home": "Start",
    "favorites": "Favoriten",
    "dictionary": "Lexikon",
    "userVocabulary": "Meine Vokabeln",
    "header": {
      "overview": "Übersicht",
      "overviewExercises": "Übersicht der Übungen",
      "cancelExercise": "Übung beenden",
      "manageSelection": "Bereiche verwalten",
      "settings": "Einstellungen",
      "wordFeedback": "Feedback geben",
      "overflowMenu": "Mehr Optionen"
    },
    "back": "Zurück",
    "plurals": "Plural",
    "noResults": "Es wurden leider keine Ergebnisse gefunden.",
    "error": {
      "unknown": "Ein unbekannter Fehler ist aufgetreten.",
      "noWifi": "Es konnte keine Verbindung mit dem Internet hergestellt werden.",
      "retryButton": "Erneut versuchen",
      "somethingWentWrong": "Ups, da ist etwas schief gelaufen!"
    },
    "rootDiscipline": "Beruf",
    "rootDisciplines": "Berufe",
    "discipline": "Modul",
    "disciplines": "Module",
    "word": "Wort",
    "words": "Wörter",
    "camera": {
      "noAuthorization": {
        "description": "Kamerazugriff erforderlich. Öffnen Sie die Einstellungen, tippen Sie auf Berechtigungen und tippen Sie auf Erlauben.",
        "settings": "Einstellungen"
      }
    }
  },
  "favorites": "Favoriten",
  "dictionary": {
    "noResults": "Es wurden leider keine Ergebnisse gefunden."
  },
  "search": {
    "enterWord": "Wort eingeben"
  },
  "settings": {
    "settings": "Einstellungen",
    "appStability": "App Stabilität",
    "appStabilityExplanation": "Durch das automatische Senden von Fehlerberichten die Lunes App verbessern",
    "version": "Version",
    "debugModal": {
      "sentry": "Send Sentry Error",
      "currentCMS": "current cms",
      "changeCMS": "change cms",
      "enableDevMode": "Enable DevMode",
      "disableDevMode": "Disable DevMode"
    }
  },
  "userVocabulary": {
    "myWords": "Meine Wörter",
    "create": "Eigenes Wort erstellen",
    "overview": {
      "list": "Vokabelliste",
      "create": "Vokabeln erstellen",
      "practice": "Vokabeln üben"
    },
    "creation": {
      "headline": "Vokabel erstellen",
      "wordPlaceholder": "Wort eingeben*",
      "articlePlaceholder": "Artikel auswählen*",
      "saveButton": "Speichern",
      "addImage": "Bild hinzufügen*",
      "addAudio": "Audio aufnehmen",
      "maxPictureUpload": "Du kannst bis zu 3 Bilder hinzufügen",
      "requiredFields": "*Pflichtfelder",
      "errorMessage": "Dies ist ein Pflichtfeld.",
      "imageErrorMessage": "Du musst 1 bis 3 Bilder auswählen"
    },
    "list": {
      "edit": "Bearbeiten",
      "finished": "Fertig",
      "noWordsYet": "Du hast noch keine eigenen Wörter erstellt.",
      "create": "Eigenes Wort erstellen",
      "confirmDeletionPart1": "Willst du das Wort",
      "confirmDeletionPart2": "wirklich löschen?",
      "confirm": "Ja, löschen",
      "cancel": "Schliessen"
    }
  }
}
