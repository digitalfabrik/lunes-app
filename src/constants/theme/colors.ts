export const COLORS = {
  primary: '#000e38',
  backgroundAccent: '#ffffff',
  background: '#f9fafb',
  text: '#3a4049',
  textSecondary: '#5b616a',
  placeholder: '#777e91',
  disabled: '#e0e4ed',
  progressIndicator: '#425bee',
  containedButtonSelected: '#3d4662',
  buttonSelectedSecondary: '#f57f7a',
  overlay: 'rgba(0, 14, 56, 0.8)',
  correct: '#0adb75',
  almostCorrect: '#ffbb4a',
  incorrect: '#ff5252',
  audioIconHighlight: '#F1635F',
  audioIconSelected: '#ffa3a3',
  shadow: 'rgba(0, 0, 0, 0.6)',
  black: '#000000',
  articleNeutral: '#72f399',
  articlePlural: '#eee12d',
  articleFeminine: '#faa7a7',
  articleMasculine: '#8cc8f3',
  link: 'blue',
  lightGreyBackground: '#e0e4ed',
}

export type Color = typeof COLORS[keyof typeof COLORS]
