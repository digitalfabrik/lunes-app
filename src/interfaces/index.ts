import { ReactElement } from 'react'

export interface ILoadingProps {
  children: ReactElement
  isLoading: boolean
}

export interface IButtonProps {
  onPress: () => void
  disabled?: boolean
  children: ReactElement
  theme?: string
  testID?: string
}
export interface IPopoverProps {
  setIsPopoverVisible: (param: boolean) => void
  isVisible: boolean
  children: ReactElement
}
