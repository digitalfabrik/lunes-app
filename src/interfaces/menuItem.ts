import { ReactElement } from 'react'

export interface IMenuItemProps {
  selected: boolean
  onPress: () => void
  icon: string
  title: string
  children: ReactElement
}
