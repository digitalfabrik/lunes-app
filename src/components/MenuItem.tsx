import React from 'react'
import { View, Text, Pressable, Image, StyleSheet } from 'react-native'
import { Arrow } from '../../assets/images'
import { COLORS } from '../constants/colors'
import { IMenuItemProps } from '../interfaces/menuItem'
import { widthPercentageToDP as wp } from 'react-native-responsive-screen'

export const styles = StyleSheet.create({
  wrapper: { paddingHorizontal: 16 },
  container: {
    justifyContent: 'space-between',
    alignSelf: 'center',
    paddingVertical: 17,
    paddingRight: 8,
    paddingLeft: 16,
    marginBottom: 8,
    flexDirection: 'row',
    alignItems: 'center',
    width: '100%',
    backgroundColor: COLORS.white,
    borderColor: COLORS.lunesBlackUltralight,
    borderWidth: 1,
    borderStyle: 'solid',
    borderRadius: 2
  },
  clickedContainer: {
    marginHorizontal: 16,
    justifyContent: 'space-between',
    alignSelf: 'center',
    paddingVertical: 17,
    paddingRight: 8,
    paddingLeft: 16,
    marginBottom: 8,
    flexDirection: 'row',
    alignItems: 'center',
    width: '100%',
    backgroundColor: COLORS.lunesBlack,
    borderColor: COLORS.white
  },
  itemTitle: {
    fontSize: wp('5%'),
    fontWeight: '600',
    letterSpacing: 0.11,
    marginBottom: 2,
    color: COLORS.lunesGreyDark,
    fontFamily: 'SourceSansPro-SemiBold'
  },
  clickedItemTitle: {
    fontSize: wp('5%'),
    fontWeight: '600',
    letterSpacing: 0.11,
    marginBottom: 2,
    color: COLORS.white,
    fontFamily: 'SourceSansPro-SemiBold'
  },
  icon: {
    justifyContent: 'center',
    marginRight: 10,
    width: wp('7%'),
    height: wp('7%')
  },
  left: { flexDirection: 'row', alignItems: 'center' }
})

const MenuItem = ({ selected, onPress, icon, title, children }: IMenuItemProps) => {
  const itemStyle = selected ? styles.clickedContainer : styles.container
  const itemTitleStyle = selected ? styles.clickedItemTitle : styles.itemTitle

  return (
    <View style={styles.wrapper}>
      <Pressable style={itemStyle} onPress={onPress}>
        <View style={styles.left}>
          <Image source={{ uri: icon }} style={styles.icon} />

          <View>
            <Text style={itemTitleStyle} testID='title'>
              {title}
            </Text>
            {children}
          </View>
        </View>

        <Arrow fill={selected ? COLORS.lunesRedLight : COLORS.lunesBlack} testID='arrow' />
      </Pressable>
    </View>
  )
}

export default MenuItem
