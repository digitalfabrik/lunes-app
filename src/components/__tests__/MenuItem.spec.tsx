import React from 'react'
import MenuItem, { styles } from '../MenuItem'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { IMenuItemProps } from '../../interfaces/menuItem'
import { Text } from 'react-native'
import { COLORS } from '../../constants/colors'

describe('Components', () => {
  describe('MenuItem', () => {
    const defaultMenuItemProps: IMenuItemProps = {
      selected: false,
      icon: '',
      title: '',
      children: <Text>description</Text>,
      onPress: () => {}
    }

    it('renders correctly across screens', () => {
      const component = shallow(<MenuItem {...defaultMenuItemProps} />)
      expect(toJson(component)).toMatchSnapshot()
    })

    it('should call onPress event', () => {
      const menuItemProps: IMenuItemProps = {
        ...defaultMenuItemProps,
        onPress: jest.fn(() => result)
      }
      const result = 'I was Pressed'

      const component = shallow(<MenuItem {...menuItemProps} />)
      expect(menuItemProps.onPress).not.toHaveBeenCalled()
      component.children().props().onPress()
      expect((menuItemProps.onPress as jest.Mock).mock.calls).toHaveLength(1)
      expect(component.children().props().onPress()).toBe(result)
    })

    it('should display title passed to it', () => {
      const menuItemProps: IMenuItemProps = {
        ...defaultMenuItemProps,
        title: 'Menu item title'
      }

      const component = shallow(<MenuItem {...menuItemProps} />)
      expect(component.find('[testID="title"]').props().children).toBe(menuItemProps.title)
    })

    it('should display image passed to it', () => {
      const menuItemProps: IMenuItemProps = {
        ...defaultMenuItemProps,
        icon: 'https://lunes.tuerantuer.org/media/images/Winkelmesser.jpeg'
      }

      const component = shallow(<MenuItem {...menuItemProps} />)
      expect(component.find('Image').prop('source')).toHaveProperty('uri', menuItemProps.icon)
    })

    it('should apply selected style when selected is true', () => {
      const menuItemProps: IMenuItemProps = {
        ...defaultMenuItemProps,
        selected: true
      }
      const containerStyle = styles.wrapper
      const titleStyle = styles.clickedItemTitle

      const component = shallow(<MenuItem {...menuItemProps} />)
      expect(component.props().style).toStrictEqual(containerStyle)
      expect(component.find('[testID="title"]').props().style).toStrictEqual(titleStyle)
    })

    it('should render children passed to it', () => {
      const menuItemProps: IMenuItemProps = {
        ...defaultMenuItemProps
      }

      const component = shallow(<MenuItem {...menuItemProps} />)
      expect(component.contains(menuItemProps.children)).toBe(true)
    })

    it('should render black arrow icon when selected is false', () => {
      const menuItemProps: IMenuItemProps = {
        ...defaultMenuItemProps
      }

      const component = shallow(<MenuItem {...menuItemProps} />)
      expect(component.find('[testID="arrow"]').props().fill).toBe(COLORS.lunesBlack)
    })

    it('should render red arrow icon when selected is true', () => {
      const menuItemProps: IMenuItemProps = {
        ...defaultMenuItemProps,
        selected: true
      }

      const component = shallow(<MenuItem {...menuItemProps} />)
      expect(component.find('[testID="arrow"]').props().fill).toBe(COLORS.lunesRedLight)
    })
  })
})
