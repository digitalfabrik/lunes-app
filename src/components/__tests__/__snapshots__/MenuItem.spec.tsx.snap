// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components MenuItem renders correctly across screens 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 16,
    }
  }
>
  <Pressable
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#e0e4ed",
        "borderRadius": 2,
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 8,
        "paddingVertical": 17,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "",
          }
        }
        style={
          Object {
            "height": 52.5,
            "justifyContent": "center",
            "marginRight": 10,
            "width": 52.5,
          }
        }
      />
      <View>
        <Text
          style={
            Object {
              "color": "#3a4049",
              "fontFamily": "SourceSansPro-SemiBold",
              "fontSize": 37.5,
              "fontWeight": "600",
              "letterSpacing": 0.11,
              "marginBottom": 2,
            }
          }
          testID="title"
        />
        <Text>
          description
        </Text>
      </View>
    </View>
    <[object Object]
      fill="#000e38"
      testID="arrow"
    />
  </Pressable>
</View>
`;
